agent/command/s3_get.go

```diff
@@ -180,75 +99,43 @@ func (c *s3get) expandParams(conf *internal.TaskConfig) error {
 // Execute expands the parameters, and then fetches the
 // resource from s3.
 func (c *s3get) Execute(ctx context.Context, comm client.Communicator, logger client.LoggerProducer, conf *internal.TaskConfig) error {
-	c.internalBuckets = conf.InternalBuckets
-
-	// expand necessary params
 	if err := c.expandParams(conf); err != nil {
 		return errors.Wrap(err, "expanding params")
 	}
 
-	// validate the params
 	if err := c.validate(); err != nil {
 		return errors.Wrap(err, "validating expanded params")
 	}
 
 	trace.SpanFromContext(ctx).SetAttributes(
 		attribute.String(s3GetBucketAttribute, c.Bucket),
-		attribute.Bool(s3GetTemporaryCredentialsAttribute, c.AwsSessionToken != ""),
+		attribute.Bool(s3GetTemporaryCredentialsAttribute, c.AWSSessionToken != ""),
 		attribute.String(s3GetRemoteFileAttribute, c.remoteFile),
 		attribute.String(s3GetExpandedRemoteFileAttribute, c.RemoteFile),
-		attribute.String(s3GetRoleARN, c.TemporaryRoleARN),
+		attribute.String(s3GetRoleARN, c.RoleARN),
+		attribute.String(s3GetAssumeRoleARN, c.assumeRoleARN),
 		attribute.Bool(s3GetInternalBucketAttribute, utility.StringSliceContains(conf.InternalBuckets, c.Bucket)),
 	)
 
-	taskData := client.TaskData{ID: conf.Task.Id, Secret: conf.Task.Secret}
-	if c.TemporaryRoleARN != "" {
-		creds, err := comm.AssumeRole(ctx, taskData, apimodels.AssumeRoleRequest{
-			RoleARN: c.TemporaryRoleARN,
-		})
-		if err != nil {
-			return errors.Wrapf(err, "getting credentials for '%s' role arn", c.TemporaryRoleARN)
-		}
-		if creds == nil {
-			return errors.Errorf("nil credentials returned for '%s' role arn", c.TemporaryRoleARN)
-		}
-		c.AwsKey = creds.AccessKeyID
-		c.AwsSecret = creds.SecretAccessKey
-		c.AwsSessionToken = creds.SessionToken
-	}
-
-	if c.temporaryUseInternalBucket {
-		creds, err := comm.S3Credentials(ctx, taskData, c.Bucket)
-		if err != nil {
-			return errors.Wrap(err, "getting S3 credentials")
-		}
-		if creds == nil {
-			return errors.New("nil credentials returned for provided role arn")
-		}
-		c.AwsKey = creds.AccessKeyID
-		c.AwsSecret = creds.SecretAccessKey
-		c.AwsSessionToken = creds.SessionToken
```

Comment:
The two if blocks above are moved inside the pail credentials provider to support longer s3 operations that might perform multiple requests